### PR TITLE
Allow floats in axis min, max, and step

### LINKED ISSUE #5: non integer range
Allow `axis` to take non integer values for min, max and step.

Probably requires own implementation of `range`. Also, we need to be careful as theoretically there exist floats that do not increase when added, resulting in an endless loop possibly.

Probably an assertion `assert(previous + step > previous)` should be added in this range.

I'll try to write a short pr for this if I don't run into any issues.

### PR DESCRIPTION
**Pull request by ModProg**
fixes #5

### CODE REVIEW

File: plotst/axis.typ [hunk 1]
```diff
@@ -2,6 +2,27 @@
...
+        while it.last() + step < max {
+          assert(it.last() + step > it.last(), message: "step size too small to increase float")
+          it.push(it.last() + step)
+        }
+      }
+      it
+    } else {
+      range(min, max, step: step)
+    }
+}
```
Comment (Pegacraft) [inline plotst/axis.typ:25]: I think this should be moved to the util.typ file. But otherwise, it looks good to me.

**Review by Pegacraft — CHANGES_REQUESTED** [review]

File: plotst/axis.typ (hunk 1 above)
Comment (ModProg) [reply plotst/axis.typ:25]: > I think this should be moved to the util.typ file. But otherwise, it looks good to me.

that caused a circular dependency

Comment (Pegacraft) [reply plotst/axis.typ:25]: I changed the util. The circular dependency shouldn't happen now

Comment (Pegacraft) [reply plotst/axis.typ:25]: I made the changes for you. Thanks for helping us out ^^ Its greatly appreciated

**Pull request merged**